eureka2-ext/eureka2-eureka1-rest-api/src/main/java/com/netflix/eureka2/eureka1/rest/query/Eureka2RegistryViewCache.java

```diff
@@ -1,124 +1,158 @@
-    private static final long DEFAULT_QUERY_TIMEOUT_MS = 30000;
 
     private final SourcedEurekaRegistry<InstanceInfo> registry;
     private final long refreshIntervalMs;
-    private final long queryTimeoutMs;
     private final Scheduler scheduler;
 
-    private final AtomicReference<Eureka2ApplicationsView> allApplicationsView = new AtomicReference<>();
+    private final AtomicReference<Eureka2FullFetchWithDeltaView> fullFetchWithDeltaView = new AtomicReference<>();
     private final Map<String, Eureka2ApplicationView> applicationMap = new ConcurrentHashMap<>();
     private final Map<String, Eureka2ApplicationsView> vipMap = new ConcurrentHashMap<>();
     private final Map<String, Eureka2ApplicationsView> secureVipMap = new ConcurrentHashMap<>();
-    private final Map<String, Eureka2InstanceView> instanceViewMap = new ConcurrentHashMap<>();
 
     public Eureka2RegistryViewCache(SourcedEurekaRegistry<InstanceInfo> registry) {
-        this(registry, DEFAULT_REFRESH_INTERVAL_MS, DEFAULT_QUERY_TIMEOUT_MS, Schedulers.computation());
+        this(registry, DEFAULT_REFRESH_INTERVAL_MS, Schedulers.computation());
     }
 
     public Eureka2RegistryViewCache(SourcedEurekaRegistry<InstanceInfo> registry,
-                                    long refreshIntervalMs,
-                                    long queryTimeoutMs) {
-        this(registry, refreshIntervalMs, queryTimeoutMs, Schedulers.computation());
+                                    long refreshIntervalMs) {
+        this(registry, refreshIntervalMs, Schedulers.computation());
     }
 
     public Eureka2RegistryViewCache(SourcedEurekaRegistry<InstanceInfo> registry,
                                     long refreshIntervalMs,
-                                    long queryTimeoutMs,
                                     Scheduler scheduler) {
         this.registry = registry;
         this.refreshIntervalMs = refreshIntervalMs;
-        this.queryTimeoutMs = queryTimeoutMs;
         this.scheduler = scheduler;
     }
 
-    public Applications findAllApplications() {
-        if (allApplicationsView.get() == null) {
-            Eureka2ApplicationsView newView = new Eureka2ApplicationsView(
-                    registry.forInterest(Interests.forFullRegistry()),
-                    refreshIntervalMs,
-                    scheduler
-            );
-            if (!allApplicationsView.compareAndSet(null, newView)) {
-                newView.close();
-            }
-        }
-        return allApplicationsView.get().latestCopy().timeout(queryTimeoutMs, TimeUnit.MILLISECONDS).take(1).toBlocking().first();
+    public Observable<Applications> findAllApplications() {
+        return getLatestFullFetch().map(REGISTRY_FETCH_TO_APPLICATIONS_FUN);
+    }
+
+    public Observable<Applications> findAllApplicationsDelta() {
+        return getLatestFullFetch().map(REGISTRY_FETCH_TO_DELTA_FUN);
     }
 
-    public Application findApplication(String appName) {
```

Comment:
We should be able to get all the non-fullFetch variants (app, vip etc) just by going to the fullFetchView, and not need to cache.